### ISSUE TITLE
Feature Request: Make C-header files compatible with C++

### ISSUE THREAD

**Issue by danielappiagyei-bc** [apache/nuttx#11230]
In a previous email thread, @xiaoxiang781216, @pkarashchenko, and I discussed making C header files compatible with c++. This is necessary so a C++ application can `#include` C headers w/o compilation issues. See this PR where I wanted to include some segger RTTI files in my c++ app and had to rename some c++ identifiers.
The conclusion was this:
> We can add a simple test like this:
  > 1. Create a context target in Makefile to generate a .cxx file which
   list and include all header files in nuttx/include
   > 2. Add the generated file to CXXSRCS
> If we just want to ensure the public nuttx header files are compatible with
c++ syntax.

I can make the changes to make the header files c++-compatible. I just want to document the feature request here for posterity

**Comment by acassis** [comment]
@danielappiagyei-bc Yes, this is a good idea! Also, when you open the PR please point to this issue to have a cross link

**Comment by danielappiagyei-bc** [comment]
> @danielappiagyei-bc Yes, this is a good idea! Also, when you open the PR please point to this issue to have a cross link

Thanks Alan, will do. Should I break up these changes into multiple PRs so that it doesn't end up being a huge one with (potentially) hundreds of files changed that make it hard to review? Not sure what our GitHub philosophy is :)

**Comment by danielappiagyei-bc** [comment]
@acassis @xiaoxiang781216 im going to start working on this so it can make it into the next release
